### DISCUSSION TITLE
Object-Oriented Features

### DISCUSSION THREAD

**Discussion by benjaminpope** [exoplanet-dev/jaxoplanet#24] [Ideas]
I just wanted to raise the question of how object oriented we want jaxoplanet & starry to be. The original versions of exoplanet & starry are object oriented with orbits and stellar/planetary map properties. It would be nice to have a similar syntax to onboard new users.

For our team's separate optics project [dLux](https://louisdesdoigts.github.io/dLux/) we have built an object-oriented Jax interface to numpyro, called [zodiax](https://louisdesdoigts.github.io/zodiax/) on top of [equinox](https://github.com/patrick-kidger/equinox). This, or another architecture, could be a nice way of replicating the original usage while maintaining good performance in Jax.

**Comment by dfm** [comment]
Thanks for bringing this up @benjaminpope!

As written, `jaxoplanet` already is object oriented with a nearly compatible interface to `exoplanet`, although there aren't really any docs yet, I know :D 

```python
from jaxoplanet import light_curves, orbits
orbit = orbits.KeplerianOrbit.init(
    period=...
)
lc = light_curves.QuadLightCurve(u1, u2).light_curve(orbit, t)
```

All of these objects interact properly with JAX as Pytrees, etc.

I don't love the idea of baking something like `zodiax` into `jaxoplanet` because I want it to provide the low-level infrastructure that can be used with _any_ JAX model building framework (or none at all!). But I'd love to keep chatting/brainstorming about these interfaces!
